### DISCUSSION TITLE
Does graphql-ws support custom (non-graphql) message types?

### DISCUSSION THREAD

**Discussion by mikecann** [enisdenjo/graphql-ws#341] [Q&A] [answered]
Suppose I want to send a custom message to the server and receive a custom message back without going through GraphQL queries / mutations / subscriptions  -> resolvers , is that possible? 

The use case is I have a game that is mostly asynchronous and uses GraphQL queries / mutations and subscriptions, but we are thinking about adding some realtime features and want to re-use the existing graphql-ws socket for the communication but use a custom message protocol.

**Comment by enisdenjo** [comment, marked as answer]
No. [Invalid messages result in immediate connection termination.](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#invalid-message)

I recommend reserving a separate WS connection for any non-GraphQL activities.

**Reply by mikecann** [reply]
Okay thanks.
